src/main/java/core/basesyntax/service/GenerateReportService.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.model.Fruit;
+import java.util.List;
+import java.util.Set;
+
+public interface GenerateReportService {
+    List<String> getResult(Set<Fruit> fruits, StorageDao storageDao);
```

Comment:
```suggestion
public interface ReportGenerator {
    List<String> generateReport (StorageDao storageDao);
```

generator should generate report about all fruits in storage. no need to pass fruit list as parameter